src/blockchain.js

```diff
@@ -36,7 +36,7 @@ Blockchain.connect = function(connectionList, opts, doneCb) {
         web3.setProvider(ethereum);
         return checkConnect(next);
       } catch (error) {
-        return next(null, false);
+        return next(new Error("could not connect to web3"), false);
```

Comment:
Don't we wanna propagate `error` here? Or is there a reason we swallow it.

Reply:
I wonder if this still works with dappConnection, what we really want is the last Error

Reply:
Yeah, that's exactly what my comment on the PR is wondering.